### ISSUE TITLE
Even more TeX-hostile characters metadata

### ISSUE THREAD

**Issue by musoke** [josephwright/gitinfo2#1]
From https://github.com/Hightor/gitinfo2/issues/24

---

The readme mentions the possibility of email addresses which are not so friendly to TeX. However, this doesn't seem to cover the possibility of `%` appearing in the git metadata. I know this may seem obscure, but I actually just ran into a commit whose author made a typo in the email address, resulting in this error:

```
(../.git/gitHeadInfo.gin)
Runaway argument?
shash={1f957a7}, lhash={1f957a772346e23ca42530467210288c67a939bf}, au\ETC.
! File ended while scanning use of \@fileswith@ptions.
<inserted text> 
                \par 
```

I ended up just "fixing" `gitHeadInfo.gin` by hand. I suspect that the simplest solution would be to update the git hooks to sanitise metadata before writing to `gitHeadInfo.gin`.

**Comment by josephwright** [comment]
I'm much more an expert on TeX than `sh`, and I think a cleaner solution would be to avoid too much scripting and rather to provide a mechanism by which data can be read verbatim. That would be a simple case of using `ltcmd`, either with a 'one data item' command or one that reads all of the data in one. I favour the former as we can then deal with anything except the char(s) we use as verb delimiters. Assuming `{}` are reasonable, then
```latex
\DeclareGitInfo{<entry>}{<data>}
```
would work well.

I can see how to do that without breaking back compatibility: we retain the existing option loading approach but deprecate in favour of data lines. @frankmittelbach reminded me to pick up this package so I'll see what he feels too.

**Comment by davidcarlisle** [comment]
yes the individual items of data from git would be better read verbatim I think. the doc says email addresses are detokenised already but I suppose it fails for `{%}`
